source/code/include/playfab/PlayFabCallRequestContainerBase.h

```diff
@@ -38,20 +39,23 @@ namespace PlayFab
         void SetRequestId(const std::string& newRequestId);
         std::string GetRequestBody() const;
         std::shared_ptr<PlayFabApiSettings> GetApiSettings() const;
+        std::shared_ptr<PlayFabAuthenticationContext> GetContext() const;
 
         /// <summary>
         /// This function is meant to handle logic of calling the error callback or success
         /// </summary>
         CallRequestContainerCallback GetCallback() const;
 
         void* GetCustomData() const;
+        virtual bool ValidateSettings() = 0;
 
     protected:
         std::string url;
         std::unordered_map<std::string, std::string> requestHeaders;
         std::string requestBody;
         std::string requestId;
-        std::shared_ptr<PlayFabApiSettings> apiSettings;
+        std::shared_ptr<PlayFabApiSettings> m_settings;
```

Comment:
>m_ [](start = 44, length = 2)

I thought we were not using the m_ as a convention.

Reply:
I had a perfect storm of simultaneous name-collisons, because all of: local variables, function input params, and this class member var; to the point where this was the easiest and most clear resolution.

---
In reply to: [343287220](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343287220) [](ancestors = 343287220)

Reply:
As I said above, "apiSettings" and "settings" both conflict with the local, and funcion param vars, so I'm inclined to keep some kind of name change here.
They're protected fields, so it doesn't really affect the public interface... (GetApiSettings and GetContext being public).
So, despite the weirdness of m_ on two variables, I'm inclined to keep it, unless you ask me to revert it...

---
In reply to: [343322062](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343322062) [](ancestors = 343322062,343287220)